maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultArtifactDescriptorReader.java

```diff
@@ -243,9 +244,10 @@ private Model loadPom( RepositorySystemSession session, ArtifactDescriptorReques
             }
             catch ( ArtifactResolutionException e )
             {
-                if ( e.getCause() instanceof ArtifactNotFoundException )
+                Throwable cause = e.getCause();
+                if ( cause instanceof ArtifactNotFoundException || cause instanceof ArtifactTransferException )
                 {
-                    missingDescriptor( session, trace, a, (Exception) e.getCause() );
+                    missingDescriptor( session, trace, a, (Exception) cause );
```

Comment:
Is the cast necessary here?

Reply:
Yes, converting a Throwable to an Exception.